### PR TITLE
Add config option for terminal cursor style

### PR DESCRIPTION
**Pull request by T1mVo**
The terminal cursor can be configured as a block, beam, underline, hollow block, or it can be completely hidden.

Release Notes:

- Added config option for terminal cursor style.

### CODE REVIEW

File: assets/settings/default.json [hunk 1]
```diff
@@ -612,7 +612,20 @@
     "toolbar": {
       // Whether to display the terminal title in its toolbar.
       "title": true
-    }
+    },
+    // Set the cursor style in the terminal.
```
Comment (SomeoneToIgnore) [inline assets/settings/default.json:616]: One thing I've noticed is that we do not update existing terminals' cursors when changing the settings?

We can do that, as
https://github.com/zed-industries/zed/blob/f330f094650a36e6f54de6e71750797bd841009c/crates/terminal_view/src/terminal_view.rs#L219 
tracks the settings change and has `self.terminal` inside it, which has `term: Arc<FairMutex<Term<ZedListener>>>,` inside itself, on which we can call `term.set_cursor_style(...)`

**Review by SomeoneToIgnore — COMMENTED** [review]
Sorry for missing this PR, but at least now I'll get the notifications inside my mail and this should go faster.

I think all we need to do is to update existing terminals' cursors to respect the settings change, and we're ready to merge.

One extra question that I'm curious about: with the default settings, Alacritty shows an filled box on focus, and an unfilled one after focus is lost.
When I use another cursor in standalone Alacritty as
```
[cursor]
style = { shape = "Beam", blinking = "Off" }
```
I get the same behavior: there's a clear distinct separation between the focused cursor and the unfocused one (always the empty box for whatever cursor shape).
But all other cursor settings for Zed seem to stay the same regardless of focus. Can we make it behave similar to standalone Alacritty and its defaults?

**Comment by SomeoneToIgnore** [comment]
Looks like it's stale after all, so I'll close it for now.

**Comment by T1mVo** [comment]
Sorry, I will get back to this next week.

**Comment by SomeoneToIgnore** [comment]
No worries and no rush, thank you for bringing this in in the first place.

**Pull request closed without merging**